libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp

```diff
@@ -0,0 +1,445 @@
+{
+    uint8_t computed_checksum = 0;
+
+    // clear buffer
+    memset(raw_data, 0, ARRAY_SIZE(raw_data));
+
+    // throttle value is computed as a polynomial defined by the config parameters
+    uint16_t throttle = linearise_throttle(thr) * THROTTLE_POSITION_FACTOR;
+
+    uint8_t idx = 0;
+
+    // set Quantity + Code + "20 bytes of records to set" + Checksum
+    computed_checksum += raw_data[idx++] = req_data.quantity = QUANTITY_SET_VALUE;
+    computed_checksum += raw_data[idx++] = req_data.code = CODE_SET_VALUE;
+    computed_checksum += raw_data[idx++] = throttle & 0xFF;
+    computed_checksum += raw_data[idx++] = (throttle >> 0x08) & 0xFF;
+
+    for (; idx < QUANTITY_SET_VALUE - 2; idx++) {
+        // 0 has no impact on checksum
+        raw_data[idx] = 0;
+    }
+    // checksum calculation
+    computed_checksum = computed_checksum % BYTE_RANGE_MAX;
+    raw_data[QUANTITY_SET_VALUE - 1] = (BYTE_RANGE_MAX - computed_checksum);
+    
+    expected_bytes = QUANTITY_ACK_SET_VALUES;
+    // write data
+    port->write(raw_data, QUANTITY_SET_VALUE);
+
+    return true;
+}
+
+
+/**
+ * @brief cyclically sends different Status requests to Hirth ECU
+ * 
+ * @return true - when successful
+ * @return false  - not implemented
+ */
+bool AP_EFI_Serial_Hirth::send_request_status() {
+
+    bool status = false;
+
+    switch (req_data.code)
+    {
+    case CODE_REQUEST_STATUS_1:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_2;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_2;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_2;
+        break;
+    case CODE_REQUEST_STATUS_2:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_3;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_3;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_3;
+        break;
+    case CODE_REQUEST_STATUS_3:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_1;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_1;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_1;
+        break;
+    default:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_1;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_1;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_1;
+        break;
+    }
+    raw_data[0] = req_data.quantity;
+    raw_data[1] = req_data.code;
+    raw_data[2] = req_data.checksum;
+
+    port->write(raw_data, QUANTITY_REQUEST_STATUS);
+
+    status = true;
+
+    return status;
+}
+
+
+/**
+ * @brief parses the response from Hirth ECU and updates the internal state instance
+ * 
+ */
+void AP_EFI_Serial_Hirth::decode_data() {
+    const uint32_t now = AP_HAL::millis();
+
+    switch (res_data.code) {
+    case CODE_REQUEST_STATUS_1: {
+        struct Record1 record1;
+
+        memcpy((uint8_t*)&record1, raw_data, sizeof(record1));
```

Comment:
```suggestion
        struct Record1 *record1 = (Record1*)raw_data;
```

.. etc and similarly for other types